### PR TITLE
Inclusion of CommonRelationships into subclasses is now performed in an inherited hook

### PR DESCRIPTION
**Pull request by texpert**
To correctly set the relationships, the inclusion of CommonRelationships into subclasses is now performed in an `inherited` hook.

Big thanks to @brian-kephart for the above change advise!

Also, the redundant `has_many :metas`has been removed from the CamaleonCms::Metas concern, as it is already defined in the CommonRelationships, which is included exactly into the same classes, which are including CamaleonCms::Metas.

### CODE REVIEW

**Comment by brian-kephart** [comment]
To define the associations from the superclass, rather than passing an object to the association, I tried to use the `inherited` hook. I couldn't quite get it to work and was in a hurry to release the fix, so I gave it up. You might be able to finish it off, though, if you want to try that approach.

My non-working code was something like this:
```ruby
# post_default.rb and term_taxonomy.rb

# this code executes in the superclass when the subclass is inherited
def self.inherited(subclass)
  # instance_eval so the code is eval'd in the context of subclass, not subclass's class, which would be Class 🤪
  subclass.instance_eval do
    include CamaleonCms::CommonRelationships
  end
end
```

Given a working version of the above idea, `CommonRelationships` wouldn't have to be altered, because the module would be added directly to subclasses, even though the code that includes it lives in the superclass to DRY things up a bit.

Like I said though, I never did get it working, and including the module on the subclasses worked immediately, so I just went that route.

**Comment by texpert** [comment]
> To define the associations from the superclass, rather than passing an object to the association, I tried to use the `inherited` hook. I couldn't quite get it to work and was in a hurry to release the fix, so I gave it up. You might be able to finish it off, though, if you want to try that approach.
> 
> My non-working code was something like this:
> 
> ```ruby
> # post_default.rb and term_taxonomy.rb
> 
> # this code executes in the superclass when the subclass is inherited
> def self.inherited(subclass)
>   # instance_eval so the code is eval'd in the context of subclass, not subclass's class, which would be Class 🤪
>   subclass.instance_eval do
>     include CamaleonCms::CommonRelationships
>   end
> end
> ```
> 
> Given a working version of the above idea, `CommonRelationships` wouldn't have to be altered, because the module would be added directly to subclasses, even though the code that includes it lives in the superclass to DRY things up a bit.
> 
> Like I said though, I never did get it working, and including the module on the subclasses worked immediately, so I just went that route.

Thank you very much, @brian-kephart!!!

It is now working correctly with `class_eval` - just added a super` call before calling the association definitions.

File: app/models/camaleon_cms/post_comment.rb [hunk 1]
```diff
@@ -1,7 +1,15 @@
 module CamaleonCms
   class PostComment < CamaleonRecord
     include CamaleonCms::Metas
-    include CamaleonCms::CommonRelationships
+
+    def self.inherited(subclass)
+      super
+
+      # instance_eval so the code is eval'd in the context of subclass, not subclass's class, which would be Class 🤪
```
Comment (brian-kephart) [inline app/models/camaleon_cms/post_comment.rb:8]: I think this change is a mistake. `PostComment` needs `CommonRelationships` and isn't inherited by other classes.

File: app/models/camaleon_cms/post_default.rb [hunk 2]
```diff
@@ -3,6 +3,15 @@ class PostDefault < CamaleonRecord
     include CamaleonCms::Metas
     include CamaleonCms::CustomFieldsRead
 
+    def self.inherited(subclass)
+      super
+
+      # instance_eval so the code is eval'd in the context of subclass, not subclass's class, which would be Class 🤪
```
Comment (brian-kephart) [inline app/models/camaleon_cms/post_default.rb:9]: Please remove my incorrect comment! 😅 Sorry about that.

File: app/apps/plugins/front_cache/front_cache_helper.rb [hunk 3]
```diff
@@ -8,6 +8,7 @@ def front_cache_front_before_load
           current_site.set_option('refresh_cache', false)
         end
 
+        # avoid cache if current visitor is logged in or development
```
Comment (brian-kephart) [inline app/apps/plugins/front_cache/front_cache_helper.rb:11]: This change is unrelated to the PR, but if it's being touched anyway can the middle clauses be
```ruby
Rails.env.development? || Rails.env.test?
```

**Review by brian-kephart — CHANGES_REQUESTED** [review]
I'm glad you got the concern working! I think `PostComment` was altered by mistake, and I made a couple trivial suggestions. Looks good, though, and I'm always happy to see spec updates.

File: app/models/camaleon_cms/post_comment.rb (hunk 1 above)
Comment (texpert) [reply app/models/camaleon_cms/post_comment.rb]: Yes, thank you!

Fixed in https://github.com/owen2345/camaleon-cms/pull/1046/commits/78dc53c2d1a8a284516957452a7d4a5129709f62

File: app/models/camaleon_cms/post_default.rb (hunk 2 above)
Comment (texpert) [reply app/models/camaleon_cms/post_default.rb]: :) NP

Fixed in https://github.com/owen2345/camaleon-cms/pull/1046/commits/2bf6e26fa8a153dd94c2661d03cf481fba0ed9ed

**Review by brian-kephart — APPROVED** [review]
Looks good! My last suggestion is to squash merge this PR.

**Comment by texpert** [comment]
> Looks good! My last suggestion is to squash merge this PR.

Done!

**Comment by texpert** [comment]
@brian-kephart, @owen2345, could you, please, craft a new release with this change?

**Comment by brian-kephart** [comment]
@texpert Does this really warrant a release? Isn't this just internal refactoring, with no behavioral changes?

**Comment by texpert** [comment]
> @texpert Does this really warrant a release? Isn't this just internal refactoring, with no behavioral changes?

Yes, it does, @brian-kephart !

I had some issues with the 2.7.2 when creating posts (but unfortunately haven't documented them), and just now have discovered it doesn't show old posts content in the admin's post editor. With the current branch everything's OK.

**Comment by brian-kephart** [comment]
Ok. 2.7.3 pushed to RubyGems.

**Comment by texpert** [comment]
Thanks a lot, @brian-kephart !

**Pull request merged**